Index: modules/ssl/mod_ssl.dep
===================================================================
--- modules/ssl/mod_ssl.dep	(revision 1620289)
+++ modules/ssl/mod_ssl.dep	(working copy)
@@ -40,7 +40,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\mod_ssl.h"\
@@ -88,7 +87,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -135,7 +133,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -182,7 +179,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -230,7 +226,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -277,7 +272,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -324,7 +318,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -371,7 +364,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -418,7 +410,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -465,7 +456,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -512,7 +502,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	"..\loggers\mod_log_config.h"\
@@ -561,7 +550,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -608,7 +596,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -655,7 +642,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -702,7 +688,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -750,7 +735,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	"..\generators\mod_status.h"\
@@ -798,7 +782,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -845,7 +828,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -892,7 +874,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -940,7 +921,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
@@ -987,7 +967,6 @@
 	"..\..\srclib\apr\include\apr_thread_rwlock.h"\
 	"..\..\srclib\openssl\inc32\openssl\conf.h"\
 	"..\..\srclib\openssl\inc32\openssl\engine.h"\
-	"..\..\srclib\openssl\inc32\openssl\store.h"\
 	"..\..\srclib\openssl\inc32\openssl\ui.h"\
 	"..\..\srclib\openssl\inc32\openssl\x509v3.h"\
 	".\ssl_expr.h"\
